xbmc/pictures/ExifParse.cpp

```diff
@@ -101,20 +103,20 @@ static void ErrNonfatal(const char* const msg, int a1, int a2);
 
 //--------------------------------------------------------------------------
 // Exif format descriptor stuff
-#define FMT_BYTE       1
-#define FMT_STRING     2
-#define FMT_USHORT     3
-#define FMT_ULONG      4
-#define FMT_URATIONAL  5
-#define FMT_SBYTE      6
-#define FMT_UNDEFINED  7
-#define FMT_SSHORT     8
-#define FMT_SLONG      9
-#define FMT_SRATIONAL 10
-#define FMT_SINGLE    11
-#define FMT_DOUBLE    12
+constexpr auto FMT_BYTE = 1;
+constexpr auto FMT_STRING = 2;
```

Comment:
Maybe worth just commenting this out (along with FMT_UNDEFINED) to get around the clang warnings.
From what i could gather, this was based on an exif implementation (https://github.com/Matthias-Wandel/jhead as far as i can tell)

the implementation we have doesnt use those 2 items, but the current implementation of jhead does actively use them both. So i guess unless someone looks to update our implementation to incorporate the upstream changes, maybe just comment them out and be done with it for now

Reply:
I think it's a good idea, also if these variables are not used they could generate new warnings for "unused variable".

Reply:
If your going to leave them as defines (and not constexpr that you were doing) might as well leave them in.
The new warnings were only being created because of the 'unusued variables' that constexpr made them. I dont believe defines have the same effect, so for simplicity, if you stick with defines, just leave it how it was, if you go to constexpr, then drop them.
I think I'd lean towards just leaving it as is, no real benefit in touching it really.

Maybe just namespace the defines (anon or actual name), probably doesn't matter to get around the initial warning.

Reply:
Actually, go with constexpr, remove the 2 unused, namespace the constexpr. I think thats the way to go

Reply:
> Maybe just namespace the defines (anon or actual name), probably doesn't matter to get around the initial warning.

#defines are not restricted to namespaces so is not going to fix initial warning (warning C4005: 'FMT_STRING': macro redefinition)

To isolate with namespace is need to convert defines to variables.

On the other hand unused variables may generate new warnings (not, unused defines).

Reply:
As is it now fixes the initial warning as the only define causing it can be removed because it is not used.

> Actually, go with constexpr, remove the 2 unused, namespace the constexpr. I think thats the way to go

This would also be valid.